src/blockdata/block.rs

```diff
@@ -209,7 +209,7 @@ impl Block {
     /// Calculate the transaction merkle root.
     pub fn merkle_root(&self) -> TxMerkleNode {
         let hashes = self.txdata.iter().map(|obj| obj.txid().as_hash());
-        bitcoin_merkle_root(hashes).into()
+        bitcoin_merkle_root(hashes).expect("txdata is never empty").into()
```

Comment:
This is not actually guaranteed - the field is public.

Either:
* Document that it panics if `txdata` is empty
* Return `Option` (I like this a bit more.)

Reply:
Returned an `Option`, done as separate patch.